lib/provisioner/plugin/utils.rb

```diff
@@ -48,6 +48,14 @@ def to_json(*a)
         end
       end
 
+      # Gets a host's SSH host key
+      def ssh_keyscan(host, type = 'rsa')
+        keytype = type == 'dsa' ? 'dss' : type
+        # TODO: find a way to do this in Ruby
+        key = `ssh-keyscan -t #{type} #{host} 2>&1 | grep #{keytype}`.split(' ')
+        return key[2]
```

Comment:
we should do some validation here to prevent storing junk in the task  (and then subsequently checking that ``nil == nil``)